crates/ruff_linter_eradicate/src/detection.rs

```diff
@@ -0,0 +1,274 @@
+    ).unwrap()
+});
+static BRACKET_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"^[()\[\]{}\s]+$").unwrap());
+static CODE_INDICATORS: &[&str] = &[
+    "(", ")", "[", "]", "{", "}", ":", "=", "%", "print", "return", "break", "continue", "import",
+];
+static CODE_KEYWORDS: Lazy<Vec<Regex>> = Lazy::new(|| {
+    vec![
+        Regex::new(r"^\s*elif\s+.*\s*:\s*$").unwrap(),
+        Regex::new(r"^\s*else\s*:\s*$").unwrap(),
+        Regex::new(r"^\s*try\s*:\s*$").unwrap(),
+        Regex::new(r"^\s*finally\s*:\s*$").unwrap(),
+        Regex::new(r"^\s*except\s+.*\s*:\s*$").unwrap(),
+    ]
+});
+static CODING_COMMENT_REGEX: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"^.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)").unwrap());
+static HASH_NUMBER: Lazy<Regex> = Lazy::new(|| Regex::new(r"#\d").unwrap());
+static MULTILINE_ASSIGNMENT_REGEX: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"^\s*([(\[]\s*)?(\w+\s*,\s*)*\w+\s*([)\]]\s*)?=.*[(\[{]$").unwrap());
+static PARTIAL_DICTIONARY_REGEX: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r#"^\s*['"]\w+['"]\s*:.+[,{]\s*$"#).unwrap());
+static PRINT_RETURN_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"^(print|return)\b\s*").unwrap());
+
+/// Returns `true` if a comment contains Python code.
+pub fn comment_contains_code(line: &str, task_tags: &[String]) -> bool {
+    let line = if let Some(line) = line.trim().strip_prefix('#') {
+        line.trim()
+    } else {
+        return false;
+    };
+
+    // Ignore non-comment related hashes (e.g., "# Issue #999").
+    if HASH_NUMBER.is_match(line) {
+        return false;
+    }
+
+    // Ignore whitelisted comments.
+    if ALLOWLIST_REGEX.is_match(line) {
+        return false;
+    }
+
+    if let Some(first) = line.split(&[' ', ':']).next() {
+        if task_tags.iter().any(|tag| tag == first) {
+            return false;
+        }
+    }
+
+    if CODING_COMMENT_REGEX.is_match(line) {
+        return false;
+    }
+
+    // Check that this is possibly code.
+    if CODE_INDICATORS.iter().all(|symbol| !line.contains(symbol)) {
+        return false;
+    }
+
+    if multiline_case(line) {
+        return true;
+    }
+
+    if CODE_KEYWORDS.iter().any(|symbol| symbol.is_match(line)) {
+        return true;
+    }
+
+    let line = PRINT_RETURN_REGEX.replace_all(line, "");
+
+    if PARTIAL_DICTIONARY_REGEX.is_match(&line) {
+        return true;
+    }
+
+    // Finally, compile the source code.
+    parser::parse_program(&line, "<filename>").is_ok()
+}
+
+/// Returns `true` if a line is probably part of some multiline code.
+fn multiline_case(line: &str) -> bool {
+    if line.ends_with('\\') {
+        return true;
+    }
+
+    if MULTILINE_ASSIGNMENT_REGEX.is_match(line) {
+        return true;
+    }
+
+    if BRACKET_REGEX.is_match(line) {
+        return true;
+    }
+
+    false
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
We may not be able to move the snapshot tests into the standalone crates, since they depend on Settings _and_ they depend on all the core linter logic.